torch/autograd/gradcheck.py

```diff
@@ -11,68 +11,68 @@
-                                    device=None) -> Tuple[torch.Tensor, ...]:
-    # Makes zero-filled tensors from outputs. If `dim` is not None, for each tensor in
-    # `output_tensors`, returns a new zero-filled tensor with height of `dim` and width of
-    # `t.numel`. Otherwise, for each tensor, returns a 1-d tensor with size (t.numel,).
+def _allocate_jacobians_with_outputs(output_tensors: Tuple, numel_input, dtype=None,
+                                     device=None) -> Tuple[torch.Tensor, ...]:
+    # Makes zero-filled tensors from outputs. If `dim` is not None, for each tensor
+    # in `output_tensors`, returns a new zero-filled tensor with height of `dim` and
+    # width of `t.numel`. Otherwise, for each tensor, returns a 1-d tensor with size
+    # (t.numel,).
     out: List[torch.Tensor] = []
     options = {"dtype": dtype, "device": device, "layout": torch.strided}
     for t in output_tensors:
-        if is_float_or_complex_tensor(t):
+        if _is_float_or_complex_tensor(t):
             out.append(t.new_zeros((numel_input, t.numel()), **options))
     return tuple(out)
 
 
-def iter_tensors(x: Union[torch.Tensor, Iterable[torch.Tensor]], only_requiring_grad: bool = False) -> Iterable[torch.Tensor]:
+def _iter_tensors(x: Union[torch.Tensor, Iterable[torch.Tensor]],
+                  only_requiring_grad: bool = False) -> Iterable[torch.Tensor]:
     if is_tensor_like(x):
         # mypy doesn't narrow type of `x` to torch.Tensor
         if x.requires_grad or not only_requiring_grad:  # type: ignore[union-attr]
             yield x  # type: ignore[misc]
     elif isinstance(x, collections.abc.Iterable) and not isinstance(x, str):
         for elem in x:
-            for result in iter_tensors(elem, only_requiring_grad):
+            for result in _iter_tensors(elem, only_requiring_grad):
                 yield result
 
 
-def iter_tensor(x_tensor):
-    # Enumerates over a tensor and provides a corresponding flat index that translates
-    # to a given rol/col in the jacobian matrix. The order is the same as as if we flatten
-    # a contiguous tensor. iter_tensor also returns a strided version of the original
-    # tensor that is able to be modified inplace. If the input tensor is strided or sparse,
-    # the returned tensor will share storage with the original. Otherwise, for opaque tensor
-    # types like mkldnn, a copy is returned.
+def _iter_tensor(x_tensor):
+    # (Only used for slow gradcheck) Returns an generator that yields at each iter:
+    #  1) a tensor: the same tensor is returned across all iterations. The tensor
+    #     is not the same as the original - it is prepared so that it can be
+    #     modified in-place. Depending on whether the input tensor is strided,
+    #     sparse, or dense, the returned tensor may or may not share storage with
+    #     the original.
+    #  2) index into the tensor in dictionary order
```

Comment:
What is dictionary order?
You mean a tuple of indices that can be used with advanced indexing?

Reply:
Dictionary order is lexical ordering i.e. (0, 0), (0, 1) (1, 0) (1, 1). Its the order in which `product(*[range(m) for m in x_tensor.size()])` yields the tuples of indices.